plugins/net_plugin/net_plugin.cpp

```diff
@@ -1789,6 +1835,52 @@ namespace eosio {
       }
    }
 
+   connection_ptr sync_manager::find_next_sync_node() {
+      fc_dlog(logger, "Number connections ${s}, sync_next_expected_num: ${e}, sync_known_lib_num: ${l}",
+              ("s", my_impl->connections.number_connections())("e", sync_next_expected_num)("l", sync_known_lib_num));
+      deque<connection_ptr> conns;
+      my_impl->connections.for_each_block_connection([&](const auto& c) {
+         if (c->should_sync_from(sync_next_expected_num, sync_known_lib_num)) {
+            conns.push_back(c);
+         }
+      });
+      if (conns.size() > sync_peer_limit) {
+         std::partial_sort(conns.begin(), conns.begin() + sync_peer_limit, conns.end(), [](const connection_ptr& lhs, const connection_ptr& rhs) {
+            return lhs->get_net_latency_ns() < rhs->get_net_latency_ns();
+         });
+      }
+
+      fc_dlog(logger, "Valid sync peers ${s}, sync_ordinal ${so}", ("s", conns.size())("so", sync_ordinal.load()));
+
+      if (conns.empty()) {
+         return {};
+      }
+      if (conns.size() == 1) { // only one available
+         ++sync_ordinal;
+         conns.front()->sync_ordinal = sync_ordinal.load();
+         return conns.front();
+      }
+
+      // keep track of which node was synced from last; round-robin among the current (sync_peer_limit) lowest latency peers
+      ++sync_ordinal;
+      // example: sync_ordinal is 6 after inc above then there may be connections with 3,4,5.
+      // Choose from the lowest sync_ordinal of the sync_peer_limit of lowest latency, note 0 means not synced from yet
+      size_t the_one = 0;
+      uint32_t lowest_ordinal = std::numeric_limits<uint32_t>::max();
+      for (size_t i = 0; i < conns.size() && i < sync_peer_limit && lowest_ordinal != 0; ++i) {
+         uint32_t sync_ord = conns[i]->sync_ordinal;
+         fc_dlog(logger, "compare sync ords, conn: ${lcid}, ord: ${l} < ${r}, latency: ${lat}us",
+                 ("lcid", conns[i]->connection_id)("l", sync_ord)("r", lowest_ordinal)("lat", conns[i]->get_net_latency_ns()/1000));
+         if (sync_ord < lowest_ordinal) {
+            the_one = i;
+            lowest_ordinal = sync_ord;
+         }
+      }
```

Comment:
`std::min_element` would compare all the conns. Also the fc_dlog is nice when debug logging.

Reply:
> `std::min_element` would compare all the conns. Also the fc_dlog is nice when debug logging.

not with the resize you just added :-). And the `std::min_element` makes the purpose of the loop clear at first sight. You could do the log with a separate `std::for_each`

Reply:
lol, true. However, it doesn't provide the short-cut out when `sync_ordinal` is 0. I can change it if you like, but personally prefer as is.

Reply:
Right it doesn't provide the shortcut. I think using the std algo make it easier to read, because you immediately know what it does, otherwise you have to decipher the loop. 
But since you prefer it as is, and as you are the one writing this code, go with your preference :-)